### PR TITLE
fix(DTFS2-6038): Dashboard sorting on other pages

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* Sort button had to be clicked on second etc pages to sort

# Changes made:
* Added sortBy to the session
* Updated tests

### CODE REVIEW

File: portal/server/controllers/dashboard/deals/index.js [hunk 1]
```diff
@@ -119,10 +119,11 @@ exports.allDeals = async (req, res) => {
   const { user } = req.session;
   const currentPage = req.params.page;
 
-  let activeSortByOrder = CONSTANTS.SORT_BY.DEFAULT;
+  let activeSortByOrder = req.session.sortBy;
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals/index.js:122]: Need a validation for session to exists, if no session exists then need `SORT_BY.DEFAULT` please.

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js [hunk 2]
```diff
@@ -85,7 +85,6 @@ context('Dashboard Deals filters - filter by multiple fields with multiple value
         .contains(ALL_DEALS[19]);
 
       dashboardDeals.next().click();
-      dashboardDeals.exporterButton().click();
```
Comment (abhi-markan) [inline e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js:88]: 1. Why these button click have been removed - Are they frivolous elements in user journey?
2. Does this PR deals just with exporter column sorting or any column, if any column then we need test cases for most common columns please.

File: portal/server/controllers/dashboard/deals/index.test.js [hunk 3]
```diff
@@ -69,6 +69,7 @@ describe('controllers/dashboard/deals', () => {
       params: { page: 1 },
       session: {
         dashboardFilters: CONSTANTS.DASHBOARD.DEFAULT_FILTERS,
+        sortBy: CONSTANTS.DASHBOARD.DEFAULT_SORT.order,
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals/index.test.js:72]: Need a test for a user journey where session does not exists please.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js (hunk 2 above)
Comment (Zainzzkk) [reply e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js:88]: 1) they were there before as it needed to reclick on 2nd page for sorting again but now it's passed in the session so not needed or breaks the test

2) only sortable column

File: portal/server/controllers/dashboard/deals/index.js (hunk 1 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.js:122]: done

File: portal/server/controllers/dashboard/deals/index.test.js (hunk 3 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.test.js:72]: done

File: e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js (hunk 2 above)
Comment (abhi-markan) [reply e2e-tests/portal/cypress/integration/journeys/dashboard/dashboard-deals/dashboard-deals-sort.spec.js:88]: If only sortable column, then do we need to add test cases for those sortable columns please?

File: portal/server/controllers/dashboard/deals/index.js [hunk 4]
```diff
@@ -119,10 +119,11 @@ exports.allDeals = async (req, res) => {
   const { user } = req.session;
   const currentPage = req.params.page;
 
-  let activeSortByOrder = CONSTANTS.SORT_BY.DEFAULT;
+  let activeSortByOrder = req.session.sortBy ? req.session.sortBy : CONSTANTS.SORT_BY.DEFAULT;
 
   if (Object.keys(req.body).length) {
     req.session.dashboardFilters = req.body;
+    req.session.sortBy = req.body.sortBy ? req.body.sortBy : CONSTANTS.SORT_BY.DEFAULT;
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals/index.js:126]: Can be replaced with nullish coalescing

```
req.body.sortBy ?? CONSTANTS.SORT_BY.DEFAULT

File: portal/server/controllers/dashboard/deals/index.test.js [hunk 5]
```diff
@@ -135,6 +136,40 @@ describe('controllers/dashboard/deals', () => {
     });
   });
 
+  describe('getAllDealsData with blank session.sortBy', () => {
+    it('should calls api.allDeals with default sort query if session.sortBy is blank', async () => {
+      mockReq.session.sortBy = null;
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals/index.test.js:141]: Please replace this with 
```
delete mockRed.session.sortBy

File: portal/server/controllers/dashboard/deals/index.test.js [hunk 6]
```diff
@@ -135,6 +136,40 @@ describe('controllers/dashboard/deals', () => {
...
+
+      await getAllDealsData(
+        'mock-token',
+        mockReq.session.user,
+        mockReq.session.dashboardFilters,
+        mockReq.params.page,
+        mockReq.session.sortBy,
```
Comment (abhi-markan) [inline portal/server/controllers/dashboard/deals/index.test.js:148]: Optional chaining perhaps, as by this point `sortBy` will not exists.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: portal/server/controllers/dashboard/deals/index.js (hunk 4 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.js:126]: done

File: portal/server/controllers/dashboard/deals/index.test.js (hunk 5 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.test.js:141]: done

File: portal/server/controllers/dashboard/deals/index.test.js (hunk 6 above)
Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.test.js:148]: but that is the point of this test? I am testing sending a blank/null sortBy to the function to check that it is replaced by the default sort

Comment (Zainzzkk) [reply portal/server/controllers/dashboard/deals/index.test.js:148]: discussed

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**